notify/slack/slack.go

```diff
@@ -183,5 +184,9 @@ func (n *Notifier) Notify(ctx context.Context, as ...*types.Alert) (bool, error)
 	// Only 5xx response codes are recoverable and 2xx codes are successful.
 	// https://api.slack.com/incoming-webhooks#handling_errors
 	// https://api.slack.com/changelog/2016-05-17-changes-to-errors-for-incoming-webhooks
-	return n.retrier.Check(resp.StatusCode, resp.Body)
+	retry, err := n.retrier.Check(resp.StatusCode, resp.Body)
+	if err != nil {
+		err = fmt.Errorf("%v on slack channel: %s", err, req.Channel)
```

Comment:
You can use `github.com/pkg/errors` to wrap the error which is a bit shorter:

```go
       retry, err := n.retrier.Check(resp.StatusCode, resp.Body)
       err = errors.Wrap(err, fmt.Sprintf("channel %q", req.Channel))
       return retry, err
```

Reply:
thanks for the tip